### PR TITLE
Fix OUTPUT_FORCE option.

### PR DESCRIPTION
**Pull request by tbohn**
This addresses the requirement that initialize_atmos() have information about the number of veg tiles in the grid cell to know how to read the forcing files (if they contain timeseries of LAI, vegcover, or albedo).  Previously VIC wasn't reading the veg lib/param files when in OUTPUT_FORCE mode, which resulted in segmentation faults when trying to access veg_lib or veg_param data structures.  Now VIC reads those files even in OUTPUT_FORCE mode.

This addresses issue #305.

### CODE REVIEW

**Comment by jhamman** [comment]
Thanks @tbohn - 

Can you show a comparison of VIC 4.1.2 with `OUTPUT_FORCE` on vs this branch?

**Comment by tbohn** [comment]
You mean plot the outputs?

**Comment by jhamman** [comment]
Yes - I think a test either plotting the outputs or numerical comparison is all we need to move on here.

**Comment by tbohn** [comment]
I just ran the two versions on a test grid cell and the results were
exactly the same - I did a diff of the files.  Is my telling you this
sufficient?

**Comment by jhamman** [comment]
@bartnijssen - This is ready for your review.

File: src/initialize_atmos.c [hunk 1]
```diff
@@ -1422,174 +1421,172 @@ void initialize_atmos(atmos_data_struct        *atmos,
...
+            veg_hist[rec][v].LAI[i] = 0;
+            while (hour < rec*global_param.dt + (i+1)*options.SNOW_STEP + global_param.starthour - hour_offset_int) {
+              idx = hour;
+              if (idx < 0) idx += 24;
+              if (local_veg_hist_data[LAI_IN][v][idx] != NODATA_VH)
+	        veg_hist[rec][v].LAI[i] = local_veg_hist_data[LAI_IN][v][idx];
+              hour++;
             }
-            if(NF>1) veg_hist[rec][v].LAI[NR] = sum / (float)NF;
+     sum += veg_hist[rec][v].LAI[i];
```
Comment (bartnijssen) [inline src/initialize_atmos.c]: Fix indentation on this line

**Comment by bartnijssen** [comment]
1 indentation fix. @jhamman : After that, please go ahead and merge.

**Comment by jhamman** [comment]
@tbohn - this just needs one update and a rebase.  Then 4.2.c is good to go.

**Comment by tbohn** [comment]
Can you be more specific about what I need to do (which github actions do I
need to take, on which branches)?

**Comment by jhamman** [comment]
@tbohn - 

This branch has merge conflicts that you need to resolve.  You can do that by merging `upstream/master` into this branch or by rebasing against the current `upstream/master`?

**Comment by tbohn** [comment]
@jhamman OK, I believe I've resolved the merge conflicts.  If you can look it over and let me know what you think, that would be great.

**Comment by jhamman** [comment]
Looks good.  Thanks @tbohn.

**Pull request merged**